### DISCUSSION TITLE
[Grafana setup] Powershell script error

### DISCUSSION THREAD

**Discussion by o0Noise0o** [grafana/grafana#63362] [General]
Dear all,

I am haveing an issue with the Setup for an windows machine in Grafana.
I am following the guide and im at step two which says:

Install the Grafana Agent for Windows
 The default scrape interval is set to 60s. Scrape intervals lower than 60s may result in increased costs (DPM > 1). To learn more, please see[ Active Series and DPM.](https://grafana.com/docs/grafana-cloud/fundamentals/active-series-and-dpm/)

Download the Powershell script - run the following command in command prompt:

curl -L https://raw.githubusercontent.com/grafana/agent/release/production/grafanacloud-install.ps1 --output grafanacloud-install.ps1 

But when i execute the instruction i get the following error:

Invoke-WebRequest : A parameter cannot be found that matches parameter name 'L'.
At line:1 char:6
+ curl -L https://raw.githubusercontent.com/grafana/agent/release/produ ...
+      ~~
    + CategoryInfo          : InvalidArgument: (:) [Invoke-WebRequest], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

Now i have googled a bit and Curl does have an -L parameter so i aint sure where it goes wrong.
Can u please advice what am i missing?

Also when i click the Grafana windows installer as alternative it redirects me to an 404 Github page.
How can this be?

**Comment by o0Noise0o** [comment]
Update so i saved the data out of https://raw.githubusercontent.com/grafana/agent/release/production/grafanacloud-install.ps1
and executed the script.

It returned the following error:

 GCLOUD_API_URL: https://integrations-api-eu-west-3.grafana.net
Checking and installing required Powershell-yaml module                                                                                                                                                                                         NuGet provider is required to continue                                                                                  PowerShellGet requires NuGet provider version '2.8.5.201' or newer to interact with NuGet-based repositories. The NuGet  provider must be available in 'C:\Program Files\PackageManagement\ProviderAssemblies' or
'C:\Users\paul.houterman\AppData\Local\PackageManagement\ProviderAssemblies'. You can also install the NuGet provider
by running 'Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force'. Do you want PowerShellGet to install
 and import the NuGet provider now?
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"): y

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): a
Downloading Grafana agent Windows Installer
Installing Grafana agent for Windows
.\grafana-agent-installer.exe : The term '.\grafana-agent-installer.exe' is not recognized as the name of a cmdlet, fun
ction, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the pat
h is correct and try again.
At C:\users\paul.houterman\Desktop\grafanacloud-install.ps1:44 char:1
+ .\grafana-agent-installer.exe /S
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (.\grafana-agent-installer.exe:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Retrieving and updating Grafana agent config
Saving and updating agent configuration file
Move-Item : Kan een deel van het pad niet vinden.
At C:\users\paul.houterman\Desktop\grafanacloud-install.ps1:74 char:5
+     Move-Item $config_file "C:\Program Files\Grafana Agent\agent-conf ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\users\paul.h...ent-config.yaml:FileInfo) [Move-Item], DirectoryNotFoundE
   xception
    + FullyQualifiedErrorId : MoveFileInfoItemIOError,Microsoft.PowerShell.Commands.MoveItemCommand

Wait for Grafana service to initialize
Restarting Grafana agent service
Stop-Service : Cannot find any service with service name 'Grafana Agent'.
At C:\users\paul.houterman\Desktop\grafanacloud-install.ps1:82 char:3
+   Stop-Service "Grafana Agent"
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Grafana Agent:String) [Stop-Service], ServiceCommandException
    + FullyQualifiedErrorId : NoServiceFoundForGivenName,Microsoft.PowerShell.Commands.StopServiceCommand

Start-Service : Cannot find any service with service name 'Grafana Agent'.
At C:\users\paul.houterman\Desktop\grafanacloud-install.ps1:83 char:3
+   Start-Service "Grafana Agent"
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Grafana Agent:String) [Start-Service], ServiceCommandException
    + FullyQualifiedErrorId : NoServiceFoundForGivenName,Microsoft.PowerShell.Commands.StartServiceCommand

Wait for Grafana service to initialize after restart
Get-Service : Cannot find any service with service name 'Grafana Agent'.
At C:\users\paul.houterman\Desktop\grafanacloud-install.ps1:90 char:3
+   Get-Service "Grafana Agent"
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Grafana Agent:String) [Get-Service], ServiceCommandException
    + FullyQualifiedErrorId : NoServiceFoundForGivenName,Microsoft.PowerShell.Commands.GetServiceCommand


Now i saw it created an Grafana agent executeable. The installer says the following:
startservice Failed 1053: 
 
The service has the start- of controlassignment not answered correctly. ( < - Translation of Dutch ) 

This is happening on multiple machines here. 



------------------------

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This is phrased as a question, and we'd like questions about grafana to go to our[ community forum](https://community.grafana.com/). Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
